test/test_torch.py

```diff
@@ -16651,16 +16650,154 @@ def test_svd_tall_some_col_maj(self, device, dtype):
+    complex_types = [torch.complex64, torch.complex128]
+    default_types = [torch.float, torch.double]
+    default_types_cuda = [torch.half, torch.float, torch.double]
+
+    def test_fn(self, device):
+        op = getattr(torch, op_str)
+        meth = getattr(torch.Tensor, op_str)
+        my_default_types = default_types
+        my_float_types = float_types
+        if self.device_type == 'cuda':
+            my_default_types = default_types_cuda
+            my_float_types = float_types_cuda
+            # angle_cuda is not implemented for torch.half
+            if op_str == 'angle':
+                my_default_types = [type for type in default_types_cuda if type != torch.half]
+                my_float_types = [type for type in float_types_cuda if type != torch.half]
+        if self.device_type in unary_ufuncs[op_str].complex_on:
+            my_float_types.extend(complex_types)
+
+        prev_def_type = torch.get_default_dtype()
+        try:
+            for in_type, def_type in product(int_types, my_default_types):
+                torch.set_default_dtype(def_type)
+                t = torch.tensor((1,), device=device, dtype=in_type)
+                self.assertEqual(op(t).dtype, def_type)
+                self.assertEqual(meth(t).dtype, def_type)
+        finally:
+            torch.set_default_dtype(prev_def_type)
+
+        for in_type in my_float_types:
+            t = torch.tensor((1,), device=device, dtype=in_type)
+            self.assertEqual(op(t).dtype, in_type)
+            self.assertEqual(meth(t).dtype, in_type)
+
+        # Test for out= code paths
+        for in_type in my_float_types + int_types:
+            if self.device_type not in unary_ufuncs[op_str].complex_on:
+                break
+            t = torch.tensor((1,), device=device, dtype=in_type)
+            for out_type in my_float_types:
+                # Temporary work around to avoid complex types casting to complex types on CUDA devices
+                if in_type in complex_types and out_type in complex_types and out_type != in_type \
+                        and self.device_type == 'cuda':
+                    continue
+                out_t = torch.empty((1,), device=device, dtype=out_type)
+                # Floating type promotion does not support out= calls for input as bool tensors
+                # and output as complex tensors
+                if (in_type == torch.bool and out_type in complex_types):
+                    self.assertRaises(RuntimeError, lambda: op(t, out=out_t))
+                else:
+                    self.assertEqual(op(t, out=out_t).dtype, out_type)
+            for out_type in int_types:
+                out_t = torch.empty((1,), device=device, dtype=out_type)
+                self.assertRaises(RuntimeError, lambda: op(t, out=out_t))
+
+        # Test for in-place code paths
+        try:
+            op_inplace = getattr(torch.Tensor, op_str + "_")
+        except AttributeError:
+            return
```

Comment:
This is to avoid any `AttributeError` like in case of `angle_` (in-place op for angle is not implemented).